src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IForEachLoopStatement.cs

```diff
@@ -788,8 +847,114 @@ static void Main()
                   ILocalReferenceExpression: num (OperationKind.LocalReferenceExpression, Type: System.Int32) (Syntax: 'num')
                   InConversion: null
                   OutConversion: null
+  NextVariables(0)
 ";
             VerifyOperationTreeForTest<ForEachStatementSyntax>(source, expectedOperationTree);
         }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact, WorkItem(17602, "https://github.com/dotnet/roslyn/issues/17602")]
+        public void IForEachLoopStatement_WithDeconstructDeclaration()
+        {
+            string source = @"
+class X
+{
+    public static void M((int, int)[] x)
+    {
+        /*<bind>*/foreach (var (a, b) in x)
```

Comment:
It may also be worth be testing with nested deconstruction-foreach: `foreach (var (a, (b, c)) in ...) ...`

Reply:
Sure, I will add a unit test.